service/src/test/java/uk/nhs/adaptors/gp2gp/DeterministicIdGeneratorService.java

```diff
@@ -0,0 +1,30 @@
+package uk.nhs.adaptors.gp2gp;
+
+import java.io.BufferedReader;
+import java.io.InputStreamReader;
+import java.util.LinkedList;
+import java.util.Queue;
+
+import lombok.SneakyThrows;
+import uk.nhs.adaptors.gp2gp.common.service.RandomIdGeneratorService;
+
+public class DeterministicIdGeneratorService extends RandomIdGeneratorService {
```

Comment:
I find it weird to say "DeterministicX _is a_ (extends) RandomX". Maybe "XStub" would be clearer?
```suggestion
public class RandomIdGeneratorStubService extends RandomIdGeneratorService {
```
or
```suggestion
public class RandomIdGeneratorServiceStub extends RandomIdGeneratorService {
```